public/calendar_reboot.js

```diff
@@ -921,151 +933,168 @@ var LocalCalendarStorage = function(params) {
-                allCalendarIds[calendarObj.state.uniqueId] = calendarObj.state.title;
-                storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
-            })
-            .catch(function () {
-                console.log("No previous calendars in storage");
-                var allCalendarIds = {};
-                allCalendarIds[calendarObj.state.uniqueId] = calendarObj.state.title;
-                storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
-            })
-            
-        return Promise.all([stateP, idsP]);
+        self.startWork();
+        return new Promise(function(resolve, reject) {
+            self.database.ref().update(updates);
+            self.endWork();
+            resolve();  // Still unsure whether I will need to use reject in here or not.
+        })
+    };
+    
+    self.setActiveById = function(calendarObjId) {
+        // Set the active calendar/object by using its Id
+        
+        var userId = self.user.uid;
+        var updates = {};
+        updates['users/' + userId + '/currentActiveCalendar/'] = calendarObjId;
+        
+        self.startWork();
+        return new Promise( function(resolve, reject) {
+            self.database.ref().update(updates);
+            self.endWork();
+            resolve();  // Reject somewhere?
+        })
 
     };
     
     self.remove = function(calendarObj) {
-        //remove an app object (like a calendar) from storage
+        // Remove an app object (like a calendar) from storage
+        // To do: Determine if this method is necessary
+        
         var uniqueId = calendarObj.state.uniqueId;
         self.removeById(uniqueId);
     };
     
     self.removeById = function(uniqueId) {
-        //remove a calendar from storage by using it's Id.
-        
-        //get the allCalendarIds object from storage
-
-        return self.getAllCalendarIds()
-            .then(function(allCalendarIds) {
-                // Delete the calendar from allCalendarIds.
-                delete allCalendarIds[uniqueId];
-                // Save that change
-                storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
-                // Remove the calendar from local storage
-                removeFromLocalStorage(toKey(uniqueId));
-                // Remove active status from the calendar
-                removeFromLocalStorage(toKey(current_active_calendar));
-                
-            })
-            .catch(function () {
-                console.log("Unable to remove calendar");
-            });
+        // Remove a calendar from storage by using it's Id.
+        // First deleting it from /user/uid/allCalendarIds
+        // Second remove permissions
+        // third removing it from calendars/
+        // removing currentActive status has it's own method
+        var userId = self.user.uid;
         
+        self.startWork();
+        return new Promise(function(resolve, reject) {
```

Comment:
This is all broken because the `.remove()` calls are asynchronous. What is going to be the order of events in this function?